### DISCUSSION TITLE
UseStrictJson with multiple tests ?

### DISCUSSION THREAD

**Discussion by fforjan** [VerifyTests/Verify#755] [General]
I'm not sure I understand how the UseStrictJson works when 2 tests from the same class are verifying different content ?
i.e. TestA must TestA.verified.json and TestB must use TestB.verified.txt ?

A different use case, we are using VerifyJson() Verify() in the same test so we would like to control the extension per test
It seems I can use UseStrictJson but I cannot undo it ?
And As i'm using mstest, I tried to configure the extension for the TestA but it didn't work ?

**Comment by SimonCropp** [comment]
@fforjan can you share the test class

**Reply by fforjan** [reply]
Yeah because I think I made a mistake somewhere, let me try to isolate the problem

**Reply by fforjan** [reply]
@SimonCropp https://github.com/fforjan/VerifyJsonIssue
on my laptop, dotnet run :
```
Failed GeneratorVerify [109 ms]
  Error Message:
   Test method JsonIssue.UnitTest1.GeneratorVerify threw exception: 
VerifyException: Directory: /Users/GeoVah/Projects/VerifyJsonIssue
Delete:
  - UnitTest1.GeneratorVerify.verified.txt
Equal:
  - Received: UnitTest1.GeneratorVerify.received.json
    Verified: UnitTest1.GeneratorVerify.verified.json
```

**Reply by fforjan** [reply]
this used to work with ApprovalTests

**Reply by fforjan** [reply]
(this code doesn't match what I"m seeing in my more complex code, i'm not sure if there is not a different issue)

**Comment by SimonCropp** [comment]
calling `Verify` multiple times in a single test method is not supported

try this

```
    [TestMethod]
    public async Task GeneratorVerify()
    {
        var json = Argon.JToken.Parse("{'HelloWord': true}");
        var log = "some log";
        await Verify(new {json, log});
    }
```
